apisix/core/config_etcd.lua

```diff
@@ -257,6 +260,30 @@ local function run_watch(premature)
 end
 
 
+local function run_watch(premature)
+    ::restart::
```

Comment:
**No! The watch routine must not be restarted!**
Because it's a stateful routine (e.g. last watch starts revision) and should start once and notify all child watchers that it started. Instead, you just wait for check_worker_th, if it exits, it means the worker process is exiting, then kill run_watch_th, and that's it.

Reply:
> What if the thread `do_run_watch` crashes?

It's not supposed to crash.
In fact, almost all timers did not have a crash guard.
And even if run_watch crashes, you cannot fix it by restarting it, 'cause it's a stateful routine.

Reply:
Frankly, restarting it in this scenario is an over-design.

Reply:
Agreed with @kingluo

Reply:
Updated, thank you for your review.❤️